src/templates.rs

```diff
@@ -37,7 +37,6 @@ pub const RECORD_TEMPLATE: &str = r#"
 {%- endfor %}
 {%- endif %}
 #[derive(Debug, PartialEq, Clone, serde::Deserialize, serde::Serialize{%- if derive_builders %}, derive_builder::Builder {%- endif %})]
-#[serde(default)]
```

Comment:
Could you check if `length(fields) == lenght(defaults)` and if it's true then keep `#[serde(default)]`.
Also keep the manual `impl Default`, something like:
```
impl Default for {{ name }} {
    fn default() -> {{ name }} {
        {{ name }} {
            {%- for f in fields %}
            {{ f }}: default_{{ name | lower }}_{{ f | lower | trim_start_matches(pat="r#") }}(),
            {%- endfor %}
        }
    }
}
```